artifact/tar/tar_test.cpp

```diff
@@ -0,0 +1,197 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+
+#include <artifact/tar/tar.hpp>
+
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <vector>
+#include <fstream>
+#include <ostream>
+#include <memory>
+
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+#include <common/processes.hpp>
+
+#include <common/testing.hpp>
+
+using namespace std;
+
+namespace io = mender::common::io;
+
+namespace tar = mender::tar;
+
+namespace processes = mender::common::processes;
+
+namespace mendertesting = mender::common::testing;
+
+class TarTestEnv : public testing::Test {
+public:
+protected:
+	static void SetUpTestSuite() {
+		string script = R"(#! /bin/sh
+
+    DIRNAME=$(dirname $0)
+
+		# Create small tar file
+		echo foobar > ${DIRNAME}/testdata
+		tar cvfz ${DIRNAME}/test.tar ${DIRNAME}/testdata
+
+		# Create large tar file
+		dd if=/dev/random of=${DIRNAME}/testinput.large bs=1M count=4
+		tar cvf ${DIRNAME}/test-large.tar ${DIRNAME}/testinput.large
+
+    # Create a corrupt tar file
+    cp ${DIRNAME}/test.tar ${DIRNAME}/test-corrupt.tar
+    dd if=/dev/random of=${DIRNAME}/test-corrupt.tar seek=10 count=5 bs=1 conv=notrunc
+
+		exit 0
+		)";
+
+		tmpdir = new mendertesting::TemporaryDirectory();
+
+		const string script_fname = tmpdir->Path() + "/test-script.sh";
+
+		std::ofstream os(script_fname.c_str(), std::ios::out);
+		os << script;
+		os.close();
+
+		std::cout << "Tmpdir: " + tmpdir->Path() + " Script fname: " + script_fname << std::endl;
+
+		int ret = chmod(script_fname.c_str(), S_IRUSR | S_IWUSR | S_IXUSR);
+		ASSERT_EQ(ret, 0);
+
+
+		processes::Process proc({script_fname});
+		auto ex_line_data = proc.GenerateLineData();
+		ASSERT_TRUE(ex_line_data);
+		EXPECT_EQ(proc.GetExitStatus(), 0) << "error message: " + ex_line_data.error().message;
+	}
+
+	static void TearDownTestSuite() {
+		delete tmpdir;
+		tmpdir = nullptr;
+	}
+
+	static mendertesting::TemporaryDirectory *tmpdir;
```

Comment:
Even though you can't rely on scope to determine the lifetime of `tmpdir`, I suggest using `unique_ptr<mendertesting::TemporaryDirectory>` and call `.reset()` above instead of `delete` followed by null pointer assignment. I know it makes no difference in this particular case, but we never know what might change later. Or be copied to somewhere else.

Reply:
For regular code yes, but for this? 
I've literally pasted this out of the `googletest` manual.

Reply:
If it made it more difficult, I could bend, but this should be the same, and actually removes one line, I think. Let's stick to the no-pointers rule as much as we can.